containers/hawkular-metrics-openshift-integration/src/main/java/org/hawkular/openshift/auth/OpenshiftAuthHandler.java

```diff
@@ -0,0 +1,81 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.openshift.auth;
+
+
+import static io.undertow.util.Headers.ORIGIN;
+import static io.undertow.util.Methods.OPTIONS;
+
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+
+/**
+ * An Undertow handler which delegates authentication to one of the supported strategies in Openshift environments.
+ *
+ * @author Thomas Segismont
+ */
+public class OpenshiftAuthHandler implements HttpHandler {
+    private static final String OPENSHIFT_OAUTH = "openshift-oauth";
+    private static final String HTPASSWD = "htpasswd";
+    private static final String DISABLED = "disabled";
+    private static final String SECURITY_OPTION_SYSPROP = "hawkular-metrics.openshift.auth-methods";
+    private static final String SECURITY_OPTION = System.getProperty(SECURITY_OPTION_SYSPROP, OPENSHIFT_OAUTH);
+
+    private final HttpHandler containerHandler;
+    private final Authenticator authenticator;
+
+    public OpenshiftAuthHandler(HttpHandler containerHandler) {
+        this.containerHandler = containerHandler;
+        if (SECURITY_OPTION.contains(OPENSHIFT_OAUTH)) {
+            authenticator = new TokenAuthenticator(containerHandler);
+        } else if (SECURITY_OPTION.contains(HTPASSWD)) {
+            authenticator = new BasicAuthenticator(containerHandler);
+        } else if (SECURITY_OPTION.equalsIgnoreCase(DISABLED)) {
+            authenticator = new DisabledAuthenticator(containerHandler);
+        } else {
+            throw new IllegalArgumentException("Unknown security option: " + SECURITY_OPTION);
+        }
+    }
+
+    @Override
+    public void handleRequest(HttpServerExchange serverExchange) throws Exception {
+        // Skip authentication if we are dealing with a CORS preflight request
+        // CORS preflight removes any user credentials and doesn't perform an actual invocation.
+        // http://www.w3.org/TR/cors/#cross-origin-request-with-preflight-0
+        if (OPTIONS.equals(serverExchange.getRequestMethod())
+                && serverExchange.getRequestHeaders().contains(ORIGIN)) {
+            containerHandler.handleRequest(serverExchange);
+            return;
+        }
+
+        // There are a few endpoint that should not be secured. If we secure the status endpoint when we cannot
+        // tell if the container is up and it will always be marked as pending
+        String path = serverExchange.getRelativePath();
+        if (path == null || path.equals("") || path.equals("/") || path.equals("/status") || path.equals
+                ("/static")) {
```

Comment:
So static resources were not displayed correctly. Because we need to check
if path starts with (instead of equals to) "/static".

The fix is ready and will update the PR (rebased) as soon as PR #495 is
merged.

2016-04-27 15:23 GMT+02:00 Thomas Segismont <tsegismo@redhat.com>:

> I believe I did but will try again.
>
> 2016-04-26 22:33 GMT+02:00 Matthew Wringe <notifications@github.com>:
>
>> In
>> containers/hawkular-metrics-openshift-integration/src/main/java/org/hawkular/openshift/auth/OpenshiftAuthHandler.java
>> <https://github.com/hawkular/hawkular-metrics/pull/481#discussion_r61158972>
>> :
>>
>> > +    @Override
>> > +    public void handleRequest(HttpServerExchange serverExchange) throws Exception {
>> > +        // Skip authentication if we are dealing with a CORS preflight request
>> > +        // CORS preflight removes any user credentials and doesn't perform an actual invocation.
>> > +        // http://www.w3.org/TR/cors/#cross-origin-request-with-preflight-0
>> > +        if (OPTIONS.equals(serverExchange.getRequestMethod())
>> > +                && serverExchange.getRequestHeaders().contains(ORIGIN)) {
>> > +            containerHandler.handleRequest(serverExchange);
>> > +            return;
>> > +        }
>> > +
>> > +        // There are a few endpoint that should not be secured. If we secure the status endpoint when we cannot
>> > +        // tell if the container is up and it will always be marked as pending
>> > +        String path = serverExchange.getRelativePath();
>> > +        if (path == null || path.equals("") || path.equals("/") || path.equals("/status") || path.equals
>> > +                ("/static")) {
>>
>> Ok, we still need to make sure that the html status page works properly
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/hawkular/hawkular-metrics/pull/481/files/6f1f68379c3a457ff34180821423e68be5469a2b#r61158972>
>>
>
>
>
> --
> Thomas Segismont
> JBoss ON Engineering Team
>



-- 
Thomas Segismont
JBoss ON Engineering Team